packages/client/tests/observableClient.spec.ts

```diff
@@ -142,24 +143,36 @@ describe("observableClient chainHead", () => {
         finalizedBlockHashes: firstChain.map((v) => v.blockHash),
       })
 
-      let [unpinned] = await mockClient.chainHead.mock.unpin.waitNextCall()
-
       let expected = [
         initialHash,
         ...firstChain.slice(0, -1).map((v) => v.blockHash),
       ]
-      expect(new Set(unpinned)).toEqual(new Set(expected))
+
+      expected.forEach((block) => {
+        expect(mockClient.chainHead.mock.hasUnpinned(block)).not.toBeTruthy()
+      })
+
+      await mockClient.chainHead.mock.unpin.waitNextCall()
+      expected.forEach((block) => {
+        expect(mockClient.chainHead.mock.hasUnpinned(block)).toBeTruthy()
+      })
 
       sendFinalized(mockClient, {
         finalizedBlockHashes: followingChain.map((v) => v.blockHash),
       })
 
-      unpinned = (await mockClient.chainHead.mock.unpin.waitNextCall())[0]
       expected = [
         firstChain.slice(-1)[0].blockHash,
         ...followingChain.slice(0, -1).map((v) => v.blockHash),
       ]
-      expect(new Set(unpinned)).toEqual(new Set(expected))
+
+      expected.forEach((block) => {
+        expect(mockClient.chainHead.mock.hasUnpinned(block)).not.toBeTruthy()
+      })
+      await mockClient.chainHead.mock.unpin.waitNextCall()
+      expected.forEach((block) => {
+        expect(mockClient.chainHead.mock.hasUnpinned(block)).toBeTruthy()
+      })
```

Comment:
Maybe we can expose `unpinnedHashes` from `MockChainHead` to make a single `expect(mockClient.chainHead.mock.unpinnedHashes).toEqual(expected)`, which will have the advantage of telling us which specific hashes are failing (instead of `expected false to be true`)

Reply:
I mean, the point of that is to make sure that they are not being unpinned before they should be unpinned.

I agree that exposing `unpinnedHashes` would be better, but we would still have to perform two checks, one before and one after. Although, we would be checking for exactly what it's in there, rather than the weird thing that we are currently doing.